var/spack/repos/builtin/packages/py-onnx/package.py

```diff
@@ -70,3 +70,16 @@ class PyOnnx(PythonPackage):
 
     # 'python_out' does not recognize dllexport_decl.
     patch("remove_dllexport_decl.patch", when="@:1.6.0")
+
+    # Switch the CMAKE_CXX_STANDARD to 17 if abseil-cpp has been built with
+    # either of those. (abseil-cpp is pulled in via protobuf)
+    patch(
+        "https://github.com/onnx/onnx/commit/1f6e43cb4d7366b2dffa7f70ae88198306e12c6c.patch?full_index=1",
+        sha256="be12f589bc4113982e4162efcdbd95835a6c161a9a7e10cd1dde026cadedf8aa",
+        when="@1.15.0 ^abseil-cpp cxxstd=17",
+    )
+    patch(
+        "https://github.com/onnx/onnx/commit/1f6e43cb4d7366b2dffa7f70ae88198306e12c6c.patch?full_index=1",
+        sha256="be12f589bc4113982e4162efcdbd95835a6c161a9a7e10cd1dde026cadedf8aa",
+        when="@1.15.0 ^abseil-cpp cxxstd=20",
```

Comment:
Does this patch only apply to 1.15.0, or should it be applied to more versions?
```suggestion
        when="@1.15: ^abseil-cpp cxxstd=20",
```
PS: I compiled `abseil-cpp cxxstd=17` and the build still succeeded without these patches with gcc-11.4.0.
Is there a specific failure to know of?

Reply:
The patch comes from `@1.16`, so the version specifier should be reversed to `@:1.15`. However, I haven't checked whether it applies cleanly for other versions.

Additionally, the failure was only introduced with #46434, because previously it got `protobuf` from `py-protobuf+cpp`, I think.

> the build still succeeded without these patches with gcc-11.4.0.
Is there a specific failure to know of?

Interesting. I could not build it with `gcc-12.3.0`. The failure I get, complains about `std::string_view` not naming a type, which is triggered by abseil, which has a typdef `using string_view = std::string_view` somewhere for builds with `c++17` and upwards. From a failed build:

```
[...]
 In file included from /home/tmadlener/work/.spack/spackages/protobuf/3.25.3/skylake-ubuntu22.04-gcc12.3.0/ovyews2/include/google/protobuf/stubs/common.h:20,
                   from /home/tmadlener/work/.spack/spackages/protobuf/3.25.3/skylake-ubuntu22.04-gcc12.3.0/ovyews2/include/google/protobuf/io/coded_stream.h:107,
                   from /home/tmadlener/work/.spack/spack-build-stage/spack-stage-py-onnx-1.15.0-jjzdiofr2fstrgn3ajblsd5gtaoxgsey/spack-src/.setuptools-cmake-build/onnx/onnx-operators-ml.pb.h:26,
                   from /home/tmadlener/work/.spack/spack-build-stage/spack-stage-py-onnx-1.15.0-jjzdiofr2fstrgn3ajblsd5gtaoxgsey/spack-src/.setuptools-cmake-build/onnx/onnx-operators-ml.pb.cc:4:
  /home/tmadlener/work/.spack/spackages/abseil-cpp/20240722.0/skylake-ubuntu22.04-gcc12.3.0/lzk33me/include/absl/strings/string_view.h:53:26: error: 'string_view' in namespace 'std' does not name a type
     53 | using string_view = std::string_view;
        |                          ^~~~~~~~~~~
  /home/tmadlener/work/.spack/spackages/abseil-cpp/20240722.0/skylake-ubuntu22.04-gcc12.3.0/lzk33me/include/absl/strings/string_view.h:53:21: note: 'std::string_view' is only available from C++17 onwards
     53 | using string_view = std::string_view;
[...]
```

Reply:
Thanks for the info. I have no objection to merge as is now.